src/CVPresenter.ts

```diff
@@ -0,0 +1,129 @@
+import { IndexedFormula, NamedNode, Literal, Namespace } from "rdflib";
+import { ns, language } from "solid-ui";
+import { findImage } from "solid-ui/lib/widgets/buttons";
+import Node from "rdflib/src/node-internal";
+import { validateHTMLColorHex } from "validate-color";
+
+export interface Role {
+  startDate?: Literal,
+  endDate: Literal,
+  dates: string,
+  orgName: string,
+  roleText: string,
+  orgHomePage?: string
+}
+export interface CVPresentation { rolesByType: {
+    PastRole: Role[];
+    CurrentRole: Role[];
+    FutureRole: Role[];
+  };
+   skills: string[];
+  }
+
+const ORG = Namespace('http://www.w3.org/ns/org#');
+
+export const typesOfRole = ['PastRole', 'CurrentRole', 'FutureRole'];
+
+export function presentCV (
+  subject: NamedNode,
+  store: IndexedFormula
+): CVPresentation {
+  const profile = subject.doc()
+  const memberships = store.each(null, ORG('member'), subject, null)
+
+  function initialArray () {
+    const ar:Role[] = []
+    return ar
+  }
+
+ const rolesByType = { PastRole: [], CurrentRole: [], FutureRole: [] }
+  /*
+  for (const t of typesOfRole) {
+   rolesByType[t] = []
+  }
+  */
+  // const now = Literal.fromValue(new Date()) // toISOString
+  let endDate
+  for (const membership of memberships) {
+    let endDate, orgHomePage, orgNameGiven, publicIdName, roleName, publicId
+     if (store.holds(membership, ns.rdf('type'), ns.solid('PastRole'))) {
+       endDate = store.any(membership as any, ns.schema('endDate'))
+     }
+
+/* Just for the record this would not be a bad way to write this code in future
+  const where = store.setDoc(profile).setFolloing(false).ns(ns).where
+  const {  startDate, endDate, orgName, roleText, dates, orgHomePage, orgNameGiven } =
+    where`
+      ${membership} schema:startDate $startDate;
+                    schema:endDate? $endDate;  # optional
+                    org:organization ?org .
+
+      ?org   schema:name ?orgNameGiven;
+            solid:publicId ?publicId;
+            schema:uri ?orgHomePage .
+
+      ?publicId schema:name ?orgName .
+     `;
+  */
+
+     // Things should have start dates but we will be very lenient in this view
+     //endDate = endDate || now
+     const startDate = store.any(membership as any, ns.schema('startDate')) as Literal
+     // const publicId = store.any(membership as any, ns.solid('publicId'))
+     const organization = store.any(membership as any, ORG('organization'))
+     if (organization) {
+       orgNameGiven = store.anyJS(organization as any, ns.schema('name'))
+       orgHomePage = store.any(organization as any, ns.schema('uri'))
```

Comment:
If we can break it down into smaller functions will have an easier time testing.